File: 1194-path-in-zigzag-labelled-binary-tree/path-in-zigzag-labelled-binary-tree.js
/**
 * @param {number} label
 * @return {number[]}
 */
var pathInZigZagTree = function(label) {
    let n = Math.floor(Math.log2(label)) 
    console.log(n);
    
    let m = label
    let arr = []
    while (n > 0) {
        let equation = Math.ceil(2**n - m/2 - 1 + 2**(n-1))
        arr.unshift(equation)
        m = equation
        n--
    }
    return arr.concat(label)
};

File: 1194-path-in-zigzag-labelled-binary-tree/path-in-zigzag-labelled-binary-tree.cpp
class Solution {
public:
    vector<int> pathInZigZagTree(int label) {
        int m = label;
        int n = (int)floor(log2(m));
        const int arrayLength = (int)floor(log2(m)); 
        vector<int> arr;
        arr.push_back(label);
        while (n > 0) {
            int equation = floor(pow(2,n) - m / 2 - 1 + pow(2,n-1));
            arr.insert(arr.begin() , equation);
            m = equation;
            n--;
        }
        
        return arr;
    }
};